specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+
+| CAT_NNN
+| Trusted self-hosted debug MAY be enabled after boot (measuring) of the affected component, to an application specific governance process.
+
+|===
+
+Guarantees the system remains attestable.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#
+| Requirement
+
+| CAT_NNN
+| Lower privilege software MUST NOT be able to measure higher privilege software.
+
+| CAT_NNN
+| Software in one domain MUST NOT be able to measure software in a different domain, without consent.
+
+|===
+
+Prevents using event counters to measure across application or privilege boundaries. Event counters can be managed by higher privileged software as part of context switching across boundaries.
+ 
+=== Global Platform TEE
 
 ==== Overview
 
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "Global platform TEE use cases"]
+image::img_ch4_gp-tee.png[]
+
+https://globalplatform.org/[Global platform] defines technical standards, interface specifications and programming models, open source firmware, and certification programmes for _trusted execution environments (TEE)_. 
+
+A TEE is an isolated environment providing security services. TEE services can be available to software on multiple Harts. For example:
+
+* Payment clients
+* DRM clients and content protection
+* Secure storage
+* User identity management
+* Attestation services
+
+The TEE model divides software into physically isolated domains:
+
+* Normal domain +
+Typically hosting a _rich OS_ (for example, RTOS or Linux), and user applications. 
+* TEE domain +
+Hosts a _TEE OS_ (domain security manager) and _trusted applications (TA)_. 
+* Root domain +
+Hosts RoT firmware, including a secure monitor.
+
+The TEE OS is primarily responsible for isolation of TA, and for providing root of trust services, within the TEE domain.
+
+The OS in Normal domain typically controls scheduling on the system, across all Harts available to it. To interact with TA services in TEE domain, the OS in Normal domain interacts with a TEE OS through a secure monitor in Root domain. 
+
+The secure monitor is responsible for context switching and isolation across domain boundaries, including event management. 
+
+For the purpose of this specification, TEE deployment models can be separated as:
+
+* Static partition TEE +
+A single TEE provides security services to Normal domain. TA are typically installed at boot by RoT FW and TEE OS, though Global Platform does also define protocols for installation of TA at runtime. System configuration and resource allocation can be mostly static, making the system more deterministic. +
+ +
+_Use case examples:_ edge devices and IoT, automation, and automotive. 
+* Virtualized TEE +
+On a virtualized system, TEE can also be virtualized. In this case a _secure partition manager_ in TEE domain is responsible for isolation of multiple TEE guests (for example, an OEM TEE and separate third party TEE). This model can also support more dynamic resource allocation. +
```

Comment:
have we considreed the case where there are multiple TEE security monitors in parallel at HS. perhaps a payment TSM and an  OEM TSM, for certification, or compatibility / portability reasons.

Reply:
This is the "virtualized TEE" use case (Arm TZ-A with S-EL2, for example).

I this case there is an SPM at HS level in TEE domain, managing multiple "guest" TEE. For example, one for OEM "stuff", one for third party "stuff" (payment clients), one for Android "stuff", etc.